doc/developer_guide/developer_guide.md

```diff
@@ -28,50 +28,86 @@ bash install.sh
-| Development Commands | used to identify problems or for testing     |
+| Group                | Usage                                   |
+|----------------------|-----------------------------------------|
+| Release Commands     | during the release                      |
+| Deployment Commands  | to deploy infrastructure onto AWS cloud |
+| Development Commands | to identify problems or for testing     |
 
 ### Release commands
 
 The following commands are used during the release AWS Codebuild job:
-- `create-vm` - creates a new AMI and VM images
-- `update-release` - updates release notes of an existing Github release
-- `start-release-build` - starts the release on AWS codebuild
+* `create-vm`: Create a new AMI and VM images.
+* `update-release`: Update release notes of an existing Github release.
+* `start-release-build`: Start the release on AWS codebuild.
+
+Script `start-test-release-build`:
+* Is usually called from github workflow `release_droid_upload_github_release_assets.yml`.
+* Requires environment variable `GH_TOKEN` to contain a valid token for access to Github.
+* Requires to specify CLI option `--upload-url`.
+
+This operation usually takes around than 1:40 hours.
 
 ### Developer commands
 
 All other commands provide a subset of the features of the release commands, and can be used to identify problems or simulate the release:
-- `export-vm` - creates a new VM image from a running EC2-Instance
-- `install-dependencies` - starts an ansible-installation onto an existing EC-2 instance
-- `reset-password` - resets password on a remote EC-2-instance via ansible
-- `setup-ec2` - starts a new EC2 instance (based on an Ubuntu AMI)
-- `setup-ec2-and-install-dependencies` - starts a new EC2 instance and install dependencies via Ansible
-- `show-aws-assets` - shows AWS entities associated with a specific keyword (called __asset-id__)
-- `start-test-release` - starts a Test Release flow
-- `make-ami-public` - Changes permissions of an existing AMI such that it becomes public
+* `export-vm`: Create a new VM image from a running EC2-Instance.
+* `install-dependencies`: Start an ansible-installation onto an existing EC-2 instance.
+* `reset-password`: Reset password on a remote EC-2-instance via ansible.
+* `setup-ec2`: Start a new EC2 instance (based on an Ubuntu AMI).
+* `setup-ec2-and-install-dependencies`: Start a new EC2 instance and install dependencies via Ansible.
+  * The script will print the required SSH login for manual inspection or interaction with the EC2 instance.
+  * The instance is kept running until the user presses Ctrl-C.
+* `show-aws-assets`: Show AWS entities associated with a specific keyword (called __asset-id__).
+* `start-test-release-build`: Start a Test Release flow.
+* `make-ami-public`: Change permissions of an existing AMI such that it becomes public.
+
+Script `start-test-release-build` requires environment variable `GH_TOKEN` to contain a valid token for access to Github.
```

Comment:
Maybe add: It creates a github release and forwards this directly to the AWS code build

Reply:
see next push